src/cryptonote_core/service_node_list.cpp

```diff
@@ -2365,7 +2365,7 @@ void proof_info::update_pubkey(const crypto::ed25519_public_key &pk)
     if (this->is_active())
     {
       assert(this->active_since_height >= 0);
-      if (static_cast<uint64_t>(this->active_since_height) >= height) return false;
+      if (static_cast<int64_t>(this->active_since_height) >= static_cast<int64_t>(height)) return false;
```

Comment:
height is unsigned integer over all blockchain data stored so it must stay uint64_t